app/src/main/java/org/schabi/newpipe/player/resolver/PlaybackResolver.java

```diff
@@ -51,36 +68,164 @@ default MediaSource buildLiveMediaSource(@NonNull final PlayerDataSource dataSou
+                        throw new IllegalArgumentException("DASH manifest generation of YouTube"
+                                + "livestreams is not yet supported");
+                    }
+                } else {
+                    // DASH manual streams from other stream services do not require manifest
+                    // generation
+                    content = stream.getContent();
+                }
+                final DashManifest dashManifest;
+                try {
+                    final ByteArrayInputStream dashManifestInput = new ByteArrayInputStream(
+                            content.getBytes(StandardCharsets.UTF_8));
+                    String baseUrl = stream.getBaseUrl();
+                    if (baseUrl == null) {
+                        baseUrl = "";
+                    }
+
+                    dashManifest = new DashManifestParser().parse(Uri.parse(baseUrl),
+                            dashManifestInput);
+                } catch (final IOException e) {
+                    throw new IOException("Error when parsing manual DASH manifest", e);
+                }
+
+                return dataSource.getDashMediaSourceFactory().setTag(metadata)
+                        .createMediaSource(dashManifest);
+            }
+        } else {
+            throw new IllegalArgumentException("Unsupported delivery type" + deliveryMethod);
+        }
+    }
+
+    /**
+     * Remove streams which are using the {@link DeliveryMethod#TORRENT torrent delivery method}.
+     * @param streamList the list of {@link Stream streams} for which you want to delete the
+     *                   torrent streams.
+     */
+    default void removeTorrentStreams(@NonNull final List<? extends Stream> streamList) {
+        if (streamList.isEmpty()) {
+            return;
+        }
+        final Iterator<? extends Stream> streamIterator = streamList.iterator();
+        while (streamIterator.hasNext()) {
+            final Stream stream = streamIterator.next();
+            if (stream.getDeliveryMethod() == DeliveryMethod.TORRENT) {
+                streamIterator.remove();
+            }
+        }
+    }
+
+    /**
+     * Remove streams which are using the {@link DeliveryMethod#TORRENT torrent delivery method}
+     * and also streams which are not URLs.
+     * @param streamList the list of {@link Stream streams} for which you want to delete the
+     *                   streams that meet the conditions mentioned above.
+     */
+    default void removeTorrentAndNonUrlStreams(@NonNull final List<? extends Stream> streamList) {
+        if (streamList.isEmpty()) {
+            return;
+        }
+        final Iterator<? extends Stream> streamIterator = streamList.iterator();
+        while (streamIterator.hasNext()) {
+            final Stream stream = streamIterator.next();
+            if (stream.getDeliveryMethod() == DeliveryMethod.TORRENT || !stream.isUrl()) {
+                streamIterator.remove();
+            }
```

Comment:
These methods are similar to the ones in `ListHelper`. Why are both versions needed? I think we can keep those in `ListHelper`.

Reply:
I removed this version.